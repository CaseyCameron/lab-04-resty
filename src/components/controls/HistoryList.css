.HistoryList {
  width: 17.5vw;
  margin: 2px;
  display: flex;
  flex-direction: column;
}
.HistoryList li {
  background-color: #204060;
  color: white;
  width: 17vw;
  font-size: 8px;
  margin: 2px;
  border-radius: 5px;
  border: 1px solid lightgray;
  text-overflow: clip;
}
.HistoryList li p {
  font-weight: bold;
  margin-bottom: 1px;
}
