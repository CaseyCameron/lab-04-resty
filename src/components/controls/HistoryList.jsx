import React from 'react';
import History from './History';

const HistoryList = ({ history }) => {
  console.log('HistoryList', history);
  const historyElements = history.map((element, index) => (
    <li key={index}>
      <History {...element} />
    </li>
  ));
  return <ul>{historyElements}</ul>
}

export default HistoryList;