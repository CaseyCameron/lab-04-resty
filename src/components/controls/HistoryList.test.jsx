import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList';

describe('Method History Container', () => {
  const histories = [
    {
      method: 'GET',
      urlValue: 'https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68c8',
    },
    {
      method: 'POST',
      urlValue: 'https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68c8',
    }
  ];

  afterEach(() => cleanup());

  it('Displays a list of method history items', () => {

    const { asFragment } = render(<HistoryList history={histories} />);
    expect(asFragment()).toMatchSnapshot();
    // const { getByText } = render(<HistoryList history={histories} />);
    // expect(getByText('GET')).toBeDefined();
  });
});