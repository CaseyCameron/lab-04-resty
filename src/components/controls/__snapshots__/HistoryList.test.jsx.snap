// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Method History Container Displays a list of method history items 1`] = `
<DocumentFragment>
  <ul
    aria-label="history-items"
  >
    <li
      data-testid="history-id"
    >
      <figure>
        <figcaption>
          <p>
            GET
          </p>
          <p>
            https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68c8
          </p>
        </figcaption>
      </figure>
    </li>
    <li
      data-testid="history-id"
    >
      <figure>
        <figcaption>
          <p>
            POST
          </p>
          <p>
            https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68c8
          </p>
        </figcaption>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;
